raiden-ts/src/raiden.ts

```diff
@@ -1471,20 +1471,29 @@ export class Raiden {
             mergeMap(async (response) => response.text()),
             map((text) => {
               const suggestions = decode(SuggestedPartners, jsonParse(text));
-              assert(suggestions.length, ['empty partner suggestion from PFS', { pfs: pfs.url }]);
+
+              if (!suggestions.length) {
+                firstResponse = []
+              }
               return suggestions;
             }),
             catchError((err) => {
               // store first error and omit to retry next pfs in list
               this.log.info('Could not fetch PFS suggested partners', pfs, err);
-              if (!firstError) firstError = err;
+              if (!firstResponse) firstResponse = err;
               return EMPTY;
             }),
           ),
         ),
         first(), // throws if no first result can be fetched/decoded
         // if first errored, throw first seen error or pass current through
-        catchError((err) => throwError(firstError ?? err)),
+        catchError((err) => {
+          if (Array.isArray(firstResponse)) {
+            return of(firstResponse);
+          } else {
+            throw(firstResponse ?? err);
```

Comment:
@andrevmatos I had to change to this from `throwError(firstError ?? err))`. Is this correct, and can you explain me how both differ?

Reply:
@palango `throw` doesn't look like an exported function from `rxjs`, and instead the `throw` operator, which should work here since rxjs catches all exceptions and errors the output, but the intended behavior here would be for you to `return throwError(...)`, which will return a new observable which errors with given value upon subscription.